ui/src/lib/helpers.tsx

```diff
@@ -593,3 +593,27 @@ export const toKebabCase = (inputString: string): string => {
         .replace(/\s+/g, ' ') // Condense longer whitespace down to one space
         .replace(/\s/g, '-'); // Replace single spaces with hyphen
 };
+
+// Fetches sitemaps of a category from the sitemap S3 bucket and constructs a sitemap index of them.
+export const getSitemapIndexXML = async (category: 'publications' | 'users'): Promise<string> => {
+    let sitemapS3Keys: string[] = [];
+    try {
+        const sitemapsRequest = await api.get('/sitemaps/paths', undefined);
+        sitemapS3Keys = sitemapsRequest.data;
+    } catch (error) {
+        console.log(error);
+    }
+    // Write an XML element for each applicable sitemap we have in S3
+    const sitemapXMLElements = sitemapS3Keys
```

Comment:
Using .filter might be cleaner and easier to understand here

```
sitemapS3Keys.filter(k => k.startsWith(`${category}/`)).map(k => `<sitemap...></sitemap>`).join('');
```

This way you aren't having to use flatmap to flatten either an array or a string (which is confusing) and the filtering is a distinct step rather than being mixed in with the mapping.

Reply:
Although I should probably ask how many sitemapS3Keys are there likely to be? Might not be ideal doing separate filter step if there's a large amount due to looping through array twice.

Reply:
This is done now!